source/initial_temperature/spherical_shell.cc

```diff
@@ -38,42 +39,55 @@ namespace aspect
-
-      // this initial condition only makes sense if a boundary temperature
-      // is prescribed. verify that it is indeed
+      // Check that a boundary temperature is prescribed
       AssertThrow (this->has_boundary_temperature(),
                    ExcMessage ("This initial condition can only be used if a boundary "
                                "temperature is prescribed."));
 
+      // This initial condition only makes sense if the geometry is derived from
+      // a spherical model (i.e. a sphere, spherical shell or chunk)
+      const GeometryModel::Interface<dim> *geometry_model = &this->get_geometry_model();
       double R1;
-      if (dynamic_cast<const GeometryModel::SphericalShell<dim>*>
-          (&this->get_geometry_model())
-          != 0)
-        R1 = dynamic_cast<const GeometryModel::SphericalShell<dim>&>
-             (this->get_geometry_model()).outer_radius();
-      else if (dynamic_cast<const GeometryModel::Sphere<dim>*>
-               (&this->get_geometry_model())
-               != 0)
-        R1 = dynamic_cast<const GeometryModel::Sphere<dim>&>
-             (this->get_geometry_model()).radius();
-      else
-        R1 = dynamic_cast<const GeometryModel::Chunk<dim>&>
-             (this->get_geometry_model()).outer_radius();
 
+      if (dynamic_cast<const GeometryModel::Sphere<dim>*>(geometry_model) != 0)
+        {
+          R1 = dynamic_cast<const GeometryModel::Sphere<dim>&>
+               (this->get_geometry_model()).radius();
+        }
+      else if (dynamic_cast<const GeometryModel::SphericalShell<dim>*>(geometry_model) != 0)
+        {
+          R1 = dynamic_cast<const GeometryModel::SphericalShell<dim>&>
+               (this->get_geometry_model()).outer_radius();
+        }
+      else if (dynamic_cast<const GeometryModel::Chunk<dim>*>(geometry_model) != 0)
+        {
+          R1 = dynamic_cast<const GeometryModel::Chunk<dim>&>
+               (this->get_geometry_model()).outer_radius();
+        }
+      else if (const GeometryModel::EllipsoidalChunk<dim> *gm = dynamic_cast<const GeometryModel::EllipsoidalChunk<dim>*> (&this->get_geometry_model()))
+        {
 
-      // s = fraction of the way from
-      // the inner to the outer
-      // boundary; 0<=s<=1
+          // TODO
+          // If the eccentricity of the EllipsoidalChunk is non-zero, the radius can vary along a boundary,
+          // but the maximal depth is the same everywhere and we could calculate a representative pressure
+          // profile. However, it requires some extra logic with ellipsoidal
+          // coordinates, so for now we only allow eccentricity zero.
+          // Using the EllipsoidalChunk with eccentricity zero can still be useful,
+          // because the domain can be non-coordinate parallel.
+
+          AssertThrow(gm->get_eccentricity() == 0.0,
+                      ExcMessage("This plugin cannot be used with a non-zero eccentricity. "));
```

Comment:
This sounds more like you want to use `ExcNotImplemented()` given the TODO above